OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -241,7 +238,8 @@ public bool HasAnyValidWeapons(in Target t, bool checkForCenterTargetingWeapons
 			foreach (var armament in Armaments)
 			{
 				var checkIsValid = checkForCenterTargetingWeapons ? armament.Weapon.TargetActorCenter : !armament.IsTraitPaused;
-				if (checkIsValid && !armament.IsTraitDisabled && armament.Weapon.IsValidAgainst(t, self.World, self))
+				var reloadingStateIsValid = checkForReloadingState ? !armament.IsReloading : true;
```

Comment:
This can simplify to `!checkForReloadingState || !armament.IsReloading`, and i'd again drop the "state" from the variable name.